lib/components/DSButton.js

```diff
@@ -1,46 +1,24 @@
 import React from 'react';
 import '../../lib/styles/styles.scss';
 
-let type = '';
-let size = '';
+const getClassNameFromProp = (prop, prefix) => (prop && `${prefix}${prop}`) || '';
+const getSizeClassName = size => getClassNameFromProp(size);
+const getTypeClassName = type => getClassNameFromProp(type);
 
-const checkType = (props) => {
-  if (props.type === 'primary') {
-    type = "ds-btn-primary";
-  } else if (props.type === 'secondary') {
-    type = "ds-btn-secondary";
-  } else if (props.type === 'success') {
-    type = "ds-btn-success";
-  } else if (props.type === 'danger') {
-    type = "ds-btn-danger";
-  } else if (props.type === 'info') {
-    type = "ds-btn-info";
-  } else if (props.type === 'ghost') {
-    type = "ds-btn-ghost";
-  } else {
-    type = "ds-btn-primary";
-  }
-  return type;
+const DSButton = ({ type, circle, icon, size, children, ...rest }) => {
```

Comment:
I get what you did but who handles the state of the component then? As in if the button is disabled for example. I feel like I either need an API so that the parent can pass that to the Button. What about click handlers? If you have a good article about this let me know.. I always get Parent <> Child relationship wrong.

Reply:
https://reactjs.org/docs/forms.html#controlled-components
A parent should manage the state of those kinda components for a single source of truth


`{...rest} handles pretty much everything for you, it essentially 'transcludes' the props you provide so like

<BSButton onClick={e => console.log('clicked')} /> will log 'clicked' when you click it because that attribute gets lumped into 'rest' and then is applied to the button element itself